client/app/scripts/liveblog-edit/posts.service.ts

```diff
@@ -359,9 +389,49 @@ const postsService = (api, $q, _userList, session) => {
     };
 
     const removePost = (post) => {
-        const deleted = { deleted: true };
+        const removeParams = { deleted: true };
+
+        const items = post.groups[1].refs;
+        const deletePromises = [];
+
+        _.each(items, (item) => {
+            switch (item.item.item_type) {
+            case 'poll': {
+                deletePromises.push(
+                    api.polls.getById(item.residRef).then((pollToDelete) => {
+                        return api.polls.remove(pollToDelete);
+                    })
+                );
+                break;
+            }
+            default: {
+                deletePromises.push(
+                    api.items.getById(item.residRef).then((itemToDelete) => {
+                        return api.items.remove(itemToDelete);
+                    })
+                );
+                break;
+            }
+            }
+        });
+
+        return $q.all(deletePromises).then(() => {
+            angular.extend(removeParams, {
+                groups: [
+                    {
+                        id: 'root',
+                        refs: [{ idRef: 'main' }],
+                        role: 'grpRole:NEP',
+                    }, {
+                        id: 'main',
+                        refs: [],
+                        role: 'grpRole:Main',
+                    },
+                ],
+            });
```

Comment:
I see that have a soft delete at the moment but I noticed that you're doing a hard delete to the items and leaving a post without items referenced. We should either have soft delete for post and keep references or do hard delete to posts also. We can discuss this during our next dev meeting.

Reply:
Sure. No problem.